integration-libs/opf/cta/core/services/opf-dynamic-cta.service.ts

```diff
@@ -0,0 +1,254 @@
+/*
+ * SPDX-FileCopyrightText: 2024 SAP Spartacus team <spartacus-team@sap.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { inject, Injectable } from '@angular/core';
+import {
+  ActiveCartFacade,
+  Cart,
+  CartAddEntrySuccessEvent,
+  CartRemoveEntrySuccessEvent,
+  CartUpdateEntrySuccessEvent,
+} from '@spartacus/cart/base/root';
+import {
+  CurrencyService,
+  EventService,
+  LanguageService,
+  Product,
+  WindowRef,
+} from '@spartacus/core';
+import {
+  CtaProductItem,
+  CtaScriptsLocation,
+  CtaScriptsRequest,
+  OpfCtaFacade,
+} from '@spartacus/opf/cta/root';
+import { OpfGlobalFunctionsFacade } from '@spartacus/opf/global-functions/root';
+import {
+  GlobalFunctionsDomain,
+  KeyValuePair,
+} from '@spartacus/opf/payment/root';
+import { CurrentProductService } from '@spartacus/storefront';
+import {
+  combineLatest,
+  map,
+  merge,
+  Observable,
+  Subscription,
+  take,
+  tap,
+} from 'rxjs';
+
+@Injectable({
+  providedIn: 'root',
```

Comment:
If we providing service in `root` it should be a part of a `root` folder not `core`.